### PR TITLE
[Device management] Session details screen (PSG-686)

### LINKED ISSUE #7077: [Device management] Session details screen
#### What would you like to do?
In the session overview screen, we should have a session details entry leading to a new screen. Only add sections for non-extended data - session name, device id, IP, latest activity.
Data should be copied to clipboard on long press.

#### Why would you like to do it?
To view details for a specific session.

#### How would you like to achieve it?
https://www.figma.com/file/jI0ZdCeKKa51BDORmXdnRU/Settings%3A-UI-Refresh?node-id=964%3A58839

### Have you considered any alternatives?

_No response_

### Additional context

_No response_

### Are you willing to provide a PR?

Yes

### PR DESCRIPTION
**Pull request by mnaturel**
## Type of change

- [x] Feature
- [ ] Bugfix
- [ ] Technical
- [ ] Other :

## Content

Adding a new screen to show session details. It can be reached from the session overview screen. For now, we only have sections for non-extended data - session name, device id, IP, latest activity.

## Motivation and context

Closes #7077 

## Tests

- Enable the new device management feature flag
- Go to Settings -> Security & Privacy -> Show all sessions (V2, WIP)
- Press the current session card
- Press the Session details entry
- Check a new screen is displayed with the correct information

## Tested devices

- [ ] Physical
- [x] Emulator
- OS version(s): Android 11

## Checklist

- [ ] Changes has been tested on an Android device or Android emulator with API 21
- [x] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#accessibility
- [x] Pull request is based on the develop branch
- [x] Pull request includes a new file under ./changelog.d. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#changelog
- [x] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [x] You've made a self review of your PR
- [ ] If you have modified the screen flow, or added new screens to the application, you have updated the test [UiAllScreensSanityTest.allScreensTest()](https://github.com/vector-im/element-android/blob/main/vector/src/androidTest/java/im/vector/app/ui/UiAllScreensSanityTest.kt#L73)

### CODE REVIEW

File: vector/src/main/java/im/vector/app/features/settings/devices/v2/details/SessionDetailsContentItem.kt [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    var description: String? = null
+
+    @EpoxyAttribute
+    var hasDivider: Boolean = true
+
+    override fun bind(holder: Holder) {
+        super.bind(holder)
+        holder.sessionDetailsContentTitle.text = title
+        holder.sessionDetailsContentDescription.text = description
+        holder.sessionDetailsContentDescription.copyOnLongClick()
```
Comment (mnaturel) [inline vector/src/main/java/im/vector/app/features/settings/devices/v2/details/SessionDetailsContentItem.kt:45]: We should be able to long press on the whole view.

File: vector/src/test/java/im/vector/app/features/settings/devices/v2/overview/GetDeviceFullInfoUseCaseTest.kt [hunk 2]
```diff
@@ -104,27 +105,33 @@ class GetDeviceFullInfoUseCaseTest {
...
         fakeActiveSessionHolder.fakeSession.fakeCryptoService.myDevicesInfoWithIdLiveData.givenAsFlow()
         fakeActiveSessionHolder.fakeSession.fakeCryptoService.cryptoDeviceInfoWithIdLiveData = MutableLiveData(Optional(null))
         fakeActiveSessionHolder.fakeSession.fakeCryptoService.cryptoDeviceInfoWithIdLiveData.givenAsFlow()
 
+        // When
         val deviceFullInfo = getDeviceFullInfoUseCase.execute(A_DEVICE_ID).firstOrNull()
 
-        deviceFullInfo shouldBeEqualTo Optional(null)
+        // Then
+        deviceFullInfo shouldBeEqualTo null
```
Comment (Florian14) [inline vector/src/test/java/im/vector/app/features/settings/devices/v2/overview/GetDeviceFullInfoUseCaseTest.kt:120]: you can call `deviceFullInfo.shouldBeNull()`

**Review by Florian14 — APPROVED** [review]
Code seems really clean, just some small remarks, not blocking

File: library/ui-strings/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -3265,6 +3265,14 @@
     <string name="device_manager_session_title">Session</string>
     <!--  Examples: Last activity Yesterday at 6PM, Last activity Aug 31 at 5:47PM -->
     <string name="device_manager_session_last_activity">Last activity %1$s</string>
+    <string name="device_manager_session_details_title">Session details</string>
+    <string name="device_manager_session_details_description">Application, device, and activity information.</string>
+    <string name="device_manager_session_details_section_session_title">Session</string>
+    <string name="device_manager_session_details_session_name">Session name</string>
+    <string name="device_manager_session_details_session_id">Session ID</string>
+    <string name="device_manager_session_details_session_last_activity">Last activity</string>
+    <string name="device_manager_session_details_section_device_title">Device</string>
```
[the remark this answers is not in the document]
Comment (mnaturel) [reply library/ui-strings/src/main/res/values/strings.xml:3274]: I will create a more general strings for this title: `device_manager_device_title`.

File: library/ui-strings/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -3265,6 +3265,14 @@
...
+    <string name="device_manager_session_details_device_ip_address">IP address</string>
```
[the remark this answers is not in the document]
Comment (mnaturel) [reply library/ui-strings/src/main/res/values/strings.xml:3273]: I renamed the existing string with content "Session ID" but I didn't do the same for session name as we don't have the same translation right now. We will see later if we want to use the same. Updated in [b36a699](https://github.com/vector-im/element-android/pull/7084/commits/b36a6997e07ee5ca018976593880c1336579df19)

**Pull request merged**